src/fromager/metrics.py

```diff
@@ -0,0 +1,80 @@
+import functools
+import logging
+import time
+import typing
+from datetime import timedelta
+
+from packaging.requirements import Requirement
+from packaging.version import Version
+
+from . import context
+
+
+def timeit(description: str) -> typing.Callable:
+    def timeit_decorator(func: typing.Callable) -> typing.Callable:
+        @functools.wraps(func)
+        def wrapper_timeit(
+            *,
+            ctx: context.WorkContext,
+            req: Requirement | None = None,
+            **kwargs: typing.Any,
+        ) -> typing.Any:
+            ctx.time_description_store[func.__name__] = description
+
+            start = time.perf_counter()
```

Comment:
I am using `datetime.now(`) here https://github.com/python-wheel-build/fromager/blob/80b8563d9cb77e342a142ef7ebd9836994a45c6a/src/fromager/wheels.py#L274 and was wondering whether we should be consistent in using the timing libraries. What do you think?

Reply:
`perf_counter` is mostly used for measuring performance:
- https://pymotw.com/3/time/ (Doug's blog)
- https://stackoverflow.com/questions/66036844/time-time-or-time-perf-counter-which-is-faster/66037261#66037261

Reply:
Ah okay!

Reply:
There are multiple clock sources. `datetime` uses wall clock as clock source. A wall clock can jump backwards and forwards in time, e.g. during DST switch, when NTP syncs the wall clock with a remote server, or when a computer is put to sleep and wakes up again.

`perf_counter` is a monotonic clock (does not go back) with a high resolution. There is also `process_time`, which has different properties than `perf_counter`. See https://docs.python.org/3/library/time.html#time.get_clock_info for all clock types. 

`perf_counter` is a good choice to measure deltas between two time stamps.